security/pkg/nodeagent/test/cert_rotation/rotation_test.go

```diff
@@ -21,63 +21,100 @@ import (
 	defer setup.TearDown()
-
 	setup.StartProxy(t)
 	start := time.Now()
-	numReq := 0
 	for {
 		code, _, err := env.HTTPGet(fmt.Sprintf("http://localhost:%d/echo", setup.OutboundListenerPort))
 		if err != nil {
 			t.Errorf("Failed in request: %v", err)
 		}
-		if code != 200 {
+		if code != httpSuccessCode {
 			t.Errorf("Unexpected status code: %d", code)
 		}
-		numReq++
-		if time.Since(start) > 4*rotateInterval {
+		if time.Since(start) > proxyRunningTime {
 			break
 		}
-		time.Sleep(100 * time.Millisecond)
-	}
-
-	stats, err := setup.ProxySetup.GetStatsMap()
-	if err == nil {
-		numSSLHandshake := stats["cluster.outbound_cluster_tls.ssl.handshake"]
-		numSSLConnError := stats[fmt.Sprintf("listener.127.0.0.1_%d.ssl.connection_error", setup.InboundListenerPort)]
-		numSSLVerifyNoCert := stats[fmt.Sprintf("listener.127.0.0.1_%d.ssl.fail_verify_no_cert", setup.InboundListenerPort)]
-		numSSLVerifyCAError := stats[fmt.Sprintf("listener.127.0.0.1_%d.ssl.fail_verify_error", setup.InboundListenerPort)]
-		numOutboundSDSUpdate := stats["cluster.outbound_cluster_tls.client_ssl_socket_factory.ssl_context_update_by_sds"]
-		numInboundSDSUpdate := stats[fmt.Sprintf("listener.127.0.0.1_%d.server_ssl_socket_factory.ssl_context_update_by_sds", setup.InboundListenerPort)]
-		// Cluster config max_requests_per_connection is set to 1, the number of requests should match
-		// the number of SSL connections. This guarantees SSL connection is using the latest TLS key/cert loaded in Envoy.
-		if numSSLHandshake != uint64(numReq) {
-			t.Errorf("Number of successful SSL handshake does not match, expect %d but get %d", numReq, numSSLHandshake)
-		}
-		if numSSLConnError != 0 {
-			t.Errorf("Number of SSL connection error: %d", numSSLConnError)
-		}
-		if numSSLVerifyNoCert != 0 {
-			t.Errorf("Number of SSL handshake failures because of missing client cert: %d", numSSLVerifyNoCert)
-		}
-		if numSSLVerifyCAError != 0 {
-			t.Errorf("Number of SSL handshake failures on CA verification: %d", numSSLVerifyCAError)
-		}
-		// Verify that there are multiple SDS updates. TLS key/cert are loaded multiple times.
-		if numOutboundSDSUpdate <= 1 {
-			t.Errorf("Number of SDS updates at outbound cluster should be greater than one, get %d", numOutboundSDSUpdate)
-		}
-		if numInboundSDSUpdate <= 1 {
-			t.Errorf("Number of SDS updates at inbound listener should be greater than one, get %d", numInboundSDSUpdate)
+		time.Sleep(sleepTime)
+		cert, err := GetInboundCert(setup.InboundListenerPort)
+		if err != nil {
+			continue
 		}
-	} else {
-		t.Errorf("cannot get Envoy stats: %v", err)
+		certSet[cert] = true
+	}
+	numInboundCertUpdate := len(certSet) - 1
+	if numInboundCertUpdate < certShouldUpdateCount {
+		t.Errorf("Number of inbound Cert Update should not be less than %d, get %d", certShouldUpdateCount,numInboundCertUpdate)
 	}
+	/* TODO: Current the stats Map are Disabled because of performance requirement so comment out the code below
```

Comment:
Please don't comment out code, I think linter will fail here anyways. Git keeps all the history anyways

Reply:
Yes, please don't comment out code. Some validations are still useful in the test.